src/main/java/com/powsybl/openloadflow/sensi/AbstractSensitivityAnalysis.java

```diff
@@ -675,12 +676,45 @@ private static void checkRegulatingTerminal(Network network, String equipmentId)
         }
     }
 
-    public List<LfSensitivityFactor> readAndCheckFactors(Network network, SensitivityFactorReader factorReader, LfNetwork lfNetwork) {
-        final List<LfSensitivityFactor> lfFactors = new ArrayList<>();
+    class SensitivityFactorHolder {
+        private final Map<String, List<LfSensitivityFactor>> additionalFactorsPerContingency = new HashMap<>();
+        private final List<LfSensitivityFactor> additionalFactorsNoContingency = new ArrayList<>();
+        private final List<LfSensitivityFactor> commonFactors = new ArrayList<>();
+
+        public List<LfSensitivityFactor> getAllFactors() {
+            List<LfSensitivityFactor> allFactors = new ArrayList<>(commonFactors);
+            allFactors.addAll(additionalFactorsNoContingency);
+            allFactors.addAll(additionalFactorsPerContingency.values().stream().flatMap(List::stream).collect(Collectors.toSet()));
+            return allFactors;
+        }
+
+        public List<LfSensitivityFactor> getFactorsForContingency(String contingencyId) {
```

Comment:
Just a detail, but I think you could avoid passing null arguments to a function to have a specific behaviour, and instead call a specific method directly. This avoids a test in the method but also avoids to give a sense to the null value (here I admit that the sense is easily guessed though!).